source/Xunit.Gherkin.Quick/ScenarioXunitHook/Scenario/ScenarioDiscoverer.cs

```diff
@@ -19,21 +19,26 @@ public ScenarioDiscoverer(IMessageSink messageSink)
             ITestMethod testMethod, 
             IAttributeInfo factAttribute)
         {
-            var feature = new FeatureDiscoveryModel(new FeatureFileRepository("*.feature")).Discover(testMethod.TestClass.Class.ToRuntimeType());
+            var featureClass = testMethod.TestClass.Class.ToRuntimeType();
+            var featureClassInfo = FeatureClassInfo.FromFeatureClassType(featureClass);
 
-            foreach (var scenario in feature.Children.OfType<global::Gherkin.Ast.Scenario>())
+            var features = new FeatureDiscoveryModel(new FeatureFileRepository(featureClassInfo.FileNameSearchPattern)).Discover(featureClass);
+            foreach (var feature in features)
             {
-                var tags = feature.GetScenarioTags(scenario.Name);
-                bool skip = feature.IsScenarioIgnored(scenario.Name);
+                foreach (var scenario in feature.Item2.Children.OfType<global::Gherkin.Ast.Scenario>())
+                {
+                    var tags = feature.Item2.GetScenarioTags(scenario.Name);
+                    bool skip = feature.Item2.IsScenarioIgnored(scenario.Name);
 
-                yield return new ScenarioXunitTestCase(
-                    _messageSink, 
-                    testMethod, 
-                    feature.Name, 
-                    scenario.Name, 
-                    tags,
-                    skip,
-                    new object[] { scenario.Name });
+                    yield return new ScenarioXunitTestCase(
+                        _messageSink, 
+                        testMethod, 
+                        feature.Item2.Name, 
```

Comment:
What is Item2 and Item1? Confusing names.

Reply:
Oh this is an issue of .net tuples. I think it would be better to have an adhoc type instead of using a tuple.